analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -24,15 +24,19 @@ namespace SonarAnalyzer.Rules.CSharp
     public sealed class EqualityOnFloatingPoint : SonarDiagnosticAnalyzer
     {
         internal const string DiagnosticId = "S1244";
-        private const string MessageFormat = "Do not check floating point {0} with exact values, use a range instead.";
+        private const string MessageFormat = "Do not check floating point {0} with exact values, use {1} instead.";
 
-        private static readonly DiagnosticDescriptor rule =
-            DescriptorFactory.Create(DiagnosticId, MessageFormat);
+        private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
         private static readonly ISet<string> EqualityOperators = new HashSet<string> { "op_Equality", "op_Inequality" };
 
+        private static readonly IDictionary<string, string> FloatingPointEpsilonDependantMembers = new Dictionary<string, string>
+        {
+            [nameof(double.NaN)] = $"{nameof(double.IsNaN)}()",
```

Comment:
Interested in understanding why.

Initializer syntax with braces is the one at the collection level, whereas the dictionary-specific syntax sugar (using `[]`) was introduced later (C#6), since collection-based syntax was considered "somewhat suboptimal" (by Microsoft...).

c.p. https://learn.microsoft.com/en-us/archive/msdn-magazine/2014/special-issue/csharp-how-csharp-6-0-simplifies-clarifies-and-condenses-your-code#dictionary-initializer

Reply:
It's ugly. There are way too many different characters needed to make it work, while it does almost the same thing.

Functionally, I don't like the `[]` as it's a collection initializer that would not throw an exception in case of a duplicated key. And for collection initialization, it doesn't make sense to use duplicated keys.